controllers/components/kserve/kserve_controller.go

```diff
@@ -19,40 +19,94 @@ package kserve
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/predicates/resources"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/reconciler"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/metadata/labels"
 )
 
-// KserveReconciler reconciles a Kserve object.
-type KserveReconciler struct {
-	client.Client
-	Scheme *runtime.Scheme
-}
+// NewComponentReconciler creates a ComponentReconciler for the Dashboard API.
+func NewComponentReconciler(ctx context.Context, mgr ctrl.Manager) error {
+	_, err := reconciler.ComponentReconcilerFor(mgr, componentsv1.KserveInstanceName, &componentsv1.Kserve{}).
+		// operands - owned
+		Owns(&corev1.Secret{}).
+		Owns(&corev1.Service{}).
+		Owns(&corev1.ConfigMap{}).
+		Owns(&corev1.ServiceAccount{}).
+		Owns(&rbacv1.Role{}).
+		Owns(&rbacv1.RoleBinding{}).
+		Owns(&rbacv1.ClusterRole{}, builder.WithPredicates(clusterRolePredicates)).
+		Owns(&rbacv1.ClusterRoleBinding{}).
+		Owns(&templatev1.Template{}, builder.WithPredicates(compareHashPredicate)).
+		Owns(&featuresv1.FeatureTracker{}).
+		Owns(&networkingv1.NetworkPolicy{}).
+		Owns(&monitoringv1.ServiceMonitor{}).
+		Owns(&admissionregistrationv1.MutatingWebhookConfiguration{}).
+		Owns(&admissionregistrationv1.ValidatingWebhookConfiguration{}).
+		Owns(&appsv1.Deployment{}, builder.WithPredicates(resources.NewDeploymentPredicate())).
+		// operands - watched
+		//
+		// By default the Watches functions adds:
+		// - an event handler mapping to a cluster scope resource identified by the
+		//   components.opendatahub.io/managed-by annotation
+		// - a predicate that check for generation change for Delete/Updates events
+		//   for to objects that have the label components.opendatahub.io/managed-by
+		//   set to the current owner
+		//
+		Watches(&extv1.CustomResourceDefinition{}).
 
-// Reconcile is part of the main kubernetes reconciliation loop which aims to
-// move the current state of the cluster closer to the desired state.
-// TODO(user): Modify the Reconcile function to compare the state specified by
-// the Kserve object against the actual cluster state, and then
-// perform operations to make the cluster state reflect the state specified by
-// the user.
-//
-// For more details, check Reconcile and its Result here:
-// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.12.2/pkg/reconcile
-func (r *KserveReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
-
-	// TODO(user): your logic here
-
-	return ctrl.Result{}, nil
-}
+		// TODO should watch these as they may be created by features stuff? Owned by FT?
+		// Not sure where to import all of them from
+		//Watches(&servingv1beta1.KnativeServing{}).
+		//Watches(&maistrav1.ServiceMeshMember{}). TODO should watch? import issue
+		//Watches(&EnvoyFilter{}).
+		//Watches(&AuthorizationPolicy{}).
+		//Watches(&Gateway{}).
```

Comment:
@lburgazzoli @zdtsw I think these are the kinds that are created by the reconciler but owned by one of the FTs that it creates (and the FTs are owned by the Kserve instance). I guess since this reconciler creates them, it should ensure their state? Like if someone/something modified or deleted one of them, I don't think we'd currently reconcile it back until reconciliation happened for some other reason (I might be wrong about this, I should try it).

In any case, I struggled with importing them (either because of dependency issues for the `maistrav1` case, or just don't know where to find the types/modules for the others.